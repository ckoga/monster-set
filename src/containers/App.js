import React, { Component } from 'react';
import logo from '../logo.svg';
import './App.scss';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { getWeapons, getArmors } from '../apiCalls';
import EquipContainer from '../containers/EquipContainer';
import Header from '../Components/Header'
import Stats from '../containers/Stats';
import { addArmors, addWeapons } from '../actions/index'


export class App extends Component {
  constructor() {
    super()
  }

  componentDidMount = () => {
    getWeapons()
      .then(data => this.props.addWeapons(data))
      .catch(err => console.log(err))

    getArmors()
      .then(data => this.props.addArmors(data))
      .catch(err => console.log(err))
  }

  render () { 
    return (
      <div className="App">
        <Header />
        <div className='below-header'>
          <Stats />
          <EquipContainer />
        </div>
      </div>
    );
  }

}

export const mapDispatchToProps = dispatch => ({
  addWeapons: weapons => dispatch( addWeapons(weapons) ),
  addArmors: armors => dispatch( addArmors(armors) )
})

export default connect(null, mapDispatchToProps)(App);
