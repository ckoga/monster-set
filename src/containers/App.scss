.App {
  background-image: linear-gradient(black, darkblue);
  // height: 300vh;
  color: white;
}

.header {
  display: flex;
  justify-content: space-between;
  height: 100px;

  h1 {
    margin: 30px 0 0 20px;
  }
}

.below-header {
  display: flex;
}
