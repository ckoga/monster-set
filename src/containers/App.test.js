import React from 'react';
import { App } from './App'; //all tests need this
import { shallow } from 'enzyme';


describe('App', () => {
  let wrapper;
  beforeEach(() => {
      wrapper = shallow(<App 
        fetchArmors={jest.fn()}
        fetchWeapons={jest.fn()} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  
})