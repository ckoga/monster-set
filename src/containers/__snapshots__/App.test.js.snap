// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h1
      className="header-h1"
    >
      Monster Set
    </h1>
    <div
      className="button-div"
    >
      <NavLink
        className="nav"
        onClick={[Function]}
        to="/"
      >
        All
      </NavLink>
      <NavLink
        className="nav"
        onClick={[Function]}
        to="/weapons"
      >
        Weapons
      </NavLink>
      <NavLink
        className="nav"
        onClick={[Function]}
        to="/armor"
      >
        Armor
      </NavLink>
      <NavLink
        className="nav"
        onClick={[Function]}
        to="/equipment-box"
      >
        Equipment Box
      </NavLink>
    </div>
  </header>
  <div
    className="below-header"
  >
    <Connect(Build) />
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/weapons"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/armor"
      render={[Function]}
    />
  </div>
</div>
`;
